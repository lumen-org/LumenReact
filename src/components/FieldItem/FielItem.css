.field-item {
    flex: 1;
    padding: 0.1rem;
    margin: 0;
    height:inherit;
    width: auto;
    overflow:hidden;
    text-align: center;
    vertical-align: middle;
}

.field-item-container {
    display: flex;
    justify-content: flex-start;
    align-items: stretch;
    align-content: stretch;
    flex-flow: row wrap;

    background-color: white;
    border: 1px solid #e4e4ee;
    margin: 0.4rem;

    box-shadow: 0 0 3px #bbbbd4;
    overflow: hidden;
    padding: 0;
}

.draggable {
    cursor: pointer;
}
