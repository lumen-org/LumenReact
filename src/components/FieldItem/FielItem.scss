@import "src/assets/styles/colors";

.field-item {
    flex: 1;
    padding: 0.1rem;
    margin: 0;
    height:inherit;
    width: auto;
    overflow:hidden;
    text-align: center;
    vertical-align: middle;
}

.field-item-container {
    display: flex;
    justify-content: flex-start;
    align-items: stretch;
    align-content: stretch;
    flex-flow: row wrap;

    background-color: $field-item-background-color;
    border: 1px solid $field-item-border;
    margin: 0.4rem;

    box-shadow: 0 0 3px $field-item-box-shadow;
    overflow: hidden;
    padding: 0;
}

.draggable {
    cursor: pointer;
}
