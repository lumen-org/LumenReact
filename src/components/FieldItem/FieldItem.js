import React, { useRef } from "react";
import "./FielItem.scss";
import CloseButton from "../Button/CloseButton";
import FieldItemModal from "../FieldItemModal/FieldItemModal";

const FieldItem = ({
  value,
  handleClick = null,
  handleClose = null,
  handleModal = null,
  isOpen = false,
  input_dom = null,
}) => {
  return (
    <div className="field-item-container draggable">
      {handleClose && <CloseButton handleClose={handleClose} />}
      <div className={"field-item"} onClick={handleClick}>
        {value}
      </div>
      <FieldItemModal title={value} isOpen={isOpen} handleClose={handleModal} />
    </div>
  );
};

export default FieldItem;
