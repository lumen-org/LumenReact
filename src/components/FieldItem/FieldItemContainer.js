import { useDrag } from "react-dnd";
import React, { useState } from "react";
import FieldItem from "./FieldItem";
import { useDispatch, useSelector } from "react-redux";
import {
  addToSpecification,
  deleteFromSpecification,
} from "../../states/specifications/actions";
import { FIELD_ITEM } from "../../constants/dragAndDropTypes";
import FieldItemModal from "../FieldItemModal";
import { selectActiveSpecificationId } from "../../states/visualizations/selector";

function FieldItemContainer({ value, fieldName = "", type = FIELD_ITEM }) {
  const item = { type: type };
  const specificationId = useSelector(selectActiveSpecificationId);

  function dispatchListItem() {
    dispatch(
      deleteFromSpecification({
        id: specificationId,
        key: fieldName,
        value: value,
      })
    );
  }

  const [isModalOpen, setIsOpen] = useState(false);
  const [anchordEl, setAnchordEl] = useState({});

  const handleModalOpen = (event) => {
    setAnchordEl(event);
    setIsOpen(true);
  };
  const handleModalClose = () => {
    setIsOpen(false);
  };
  // to hook into specifications actions
  const dispatch = useDispatch();

  // Thats what is called a hook! ;)
  // which sadly requires the hook into redux above :/
  const [, drag] = useDrag({
    item,
    end(item, monitor) {
      const dropResult = monitor.getDropResult();
      if (item && dropResult) {
        if (fieldName) {
          dispatchListItem();
        }
        dispatch(
          addToSpecification({
            id: specificationId,
            key: dropResult.result,
            value: value,
          })
        );
      } else if (fieldName) {
        dispatchListItem();
      }
    },
  });

  return (
    <div ref={drag}>
      {fieldName ? (
        <FieldItem
          value={value}
          handleFieldItemClose={() => dispatchListItem()}
          handleModalOpen={handleModalOpen}
        />
      ) : (
        <FieldItem value={value} />
      )}
      {isModalOpen && (
        <FieldItemModal
          isOpen={isModalOpen}
          handleModalClose={handleModalClose}
          title={value}
          anchorEl={anchordEl}
        />
      )}
    </div>
  );
}

export default FieldItemContainer;
