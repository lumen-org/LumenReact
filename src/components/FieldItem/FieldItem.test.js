import React from "react";
import { render } from "@testing-library/react";
import FieldItem from "./FieldItem";



test("render value", () => {
  const { getByText } = render(<FieldItem value={"sepal_width"}/>);
  expect(getByText("sepal_width")).toBeInTheDocument();
});