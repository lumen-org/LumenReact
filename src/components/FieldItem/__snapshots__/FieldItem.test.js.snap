// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field.js Functionality Test render normally 1`] = `
<div
  className="field-item-container draggable"
>
  <div>
    CloseButton
  </div>
  <div
    className="field-item"
    onClick={[Function]}
  >
    sepal_width
  </div>
</div>
`;
